indico/modules/events/api.py

```diff
@@ -467,14 +487,18 @@ def _build_session_api_data(self, session_):
                 'slotTitle': block.title,
                 'address': block.address,
                 'conveners': map(self._serialize_convener, block.person_links)
-            })
+            }
+            if session_access_list:
+                block_data['allowed'] = session_access_list
+            data.append(block_data)
         return data
 
     def _build_contribution_api_data(self, contrib, include_subcontribs=True):
         data = {
             '_type': 'Contribution',
             '_fossil': self.fossils_mapping['contribution'].get(self._detail_level, None),
-            'id': contrib.legacy_mapping.legacy_contribution_id if contrib.legacy_mapping else contrib.id,
+            'id': contrib.legacy_mapping.legacy_contribution_id if contrib.legacy_mapping else contrib.friendly_id,
```

Comment:
`unicode(contrib.friendly_id)` so we don't sometimes return a string and sometimes an integer here